chevah/compat
# [#5226] Minor test fixes related to Windows 2019.

Scope
=====

`TestSystemUsers.test_getHomeFolder_posix` fails for home dirs with ending slash. This is because in salt-master's pillars I've updated buildslave's home dir to include the ending (forward or backward) slash in order to minimise the number of if clauses needed for Windows support in #5185. Eg. https://chevah.com/buildbot/builders/python-package-rhel-7/builds/10/steps/test_1/logs/stdio.

Also, `TestLocalFilesystemLocked.test_getAttributes_link_file_outside` fails on Windows 2019. Eg. https://chevah.com/buildbot/builders/python-package-win-2019/builds/10/steps/test_1/logs/stdio


Changes
=======

On POSIX systems, strip `/` from the end of a home dir when getting it.

On Windows, check for `c:\windows\system32\xcopy.exe` instead of `c:\windows\explorer.exe`, which is missing from Windows Server Core.

Also adapted a couple of elevated Windows tests for Windows 2019.


How to try and test the changes
===============================

reviewers: @adiroiban 

Please double-check my changes.

Run the tests.
